scripts/new_python_based_image.py

```diff
@@ -0,0 +1,555 @@
+import argparse
+import os
+import shutil
+import sys
+import platform
+import subprocess
+import logging
+import re
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def configure_logger(log_level):
+    """
+    Configures the logging settings based on the provided log level.
+
+    Args:
+        log_level (str): The logging level to set (e.g., 'INFO', 'DEBUG').
+    """
+    level = getattr(logging, log_level.upper(), logging.INFO)
+    logging.basicConfig(
+        level=level,
+        format="[%(levelname)s] %(asctime)s: %(message)s",
+        datefmt="%H:%M:%S"
+    )
+    LOGGER.setLevel(log_level)
+
+
+class Args:
+    """
```

Comment:
Consider making this a https://docs.python.org/3/library/dataclasses.html, it is significantly less verbose that way.